src/pages/requests/index.tsx

```diff
@@ -1,29 +1,45 @@
+import { useMemo } from "react";
 import { useTranslation } from "react-i18next";
+import { Layer, Source } from "react-map-gl";
+import { FeatureCollection, GeoJsonProperties, Geometry } from "geojson";
 
-import { useLocationsQuery, useAidRequestQuery } from "../../others/contexts/api";
+import { useLocationsQuery, useAidRequestQuery, useSuppliesQuery } from "../../others/contexts/api";
 import { Layout } from "../../others/components/Layout";
 import { Map } from "../../others/components/map/Map";
-
 import { Header } from "../../others/components/Header";
 import { Main } from "../../others/components/Main";
 import { Sidebar } from "../../others/components/Sidebar";
 import { MultiTab } from "../../others/components/MultiTab";
 import { CollapsibleTable } from "../../others/components/CollapsibleList";
-import { Layer, Source } from "react-map-gl";
 import { layerStyle } from "../../others/components/map/CircleLayerStyle";
 import { aidRequestsFixture } from "../../others/fixtures/request.fixture";
-import { FeatureCollection, GeoJsonProperties, Geometry } from "geojson";
+import { decodeAidRequest, DecodedAidRequest } from "../../others/helpers/decode-aid-requests";
+import { groupByLocation } from "../../others/helpers/group-aid-requests";
+import { assignTotal } from "../../others/helpers/assign-total";
 
 export function Requests() {
   const { t } = useTranslation();
   const { data: cities } = useLocationsQuery();
+  const { data: supplies } = useSuppliesQuery();
   const { data: aidRequests } = useAidRequestQuery();
 
+  const decodedAidRequests = useMemo(() => {
+    if (cities && supplies && aidRequests) {
+      const groupedByLocation = groupByLocation(aidRequests);
+      const groupedByLocationWithTotal = groupedByLocation.map(assignTotal);
+
+      console.log("groupedByLocationWithTotal", groupedByLocationWithTotal);
+      return {
+        groupedByLocation: groupedByLocationWithTotal.map((req) => decodeAidRequest(req, cities, supplies)),
+      };
+    } else return [];
+  }, [aidRequests, supplies, aidRequests]);
+
   if (!cities) {
     return <Layout header={<Header />}>{/* <Loader /> */}</Layout>;
   }
 
-  console.log("These are the aid requests:", aidRequests);
+  console.log("These are the decoded:", decodedAidRequests);
```

Comment:
What do you think of "These are the decoded aid requests:"?

Reply:
That was temporary, removed.